osfoffline/client/osf.py

```diff
@@ -59,9 +64,38 @@ def load(cls, request_session, *args):
                 resp = request_session.get(data['links']['next'], params={'page[size]': 250})
                 data = resp.json()
                 l.extend(data['data'])
-            return [cls(request_session, item) for item in l]
-        return cls(request_session, data['data'])
-
+            return [cls.from_data(request_session, item) for item in l]
+        return cls.from_data(request_session, data['data'])
+
+    def _fetch_related(self, relationship, data):
+        items = data['data']
+        for item in items:
+            yield item
+        if data['links'].get('next'):
+            for item in self.fetch_related(
+                    self.request_session,
+                    relationship
```

Comment:
This looks like it will go into an infinite loop as no page is provided to `fetch_related`